aiogoogle/client.py

```diff
@@ -141,26 +141,35 @@ async def list_api(self, name, preferred=None, fields=None):
         )
         return await self.as_anon(request)
 
-    async def discover(self, api_name, api_version=None, validate=False):
-        """ 
+    async def discover(self, api_name, api_version=None, validate=False, *, api_endpoint_url_fmt=None):
+        """
         Donwloads a discovery document from Google's Discovery Service V1 and sets it a ``aiogoogle.resource.GoogleAPI``
 
         Note:
 
             It is recommended that you explicitly specify an API version.
-            
+
             When you leave the API version as None, Aiogoogle uses the ``list_api`` method to search for the best fit version of the given API name.
-            
+
             This will result in sending two http requests instead of just one.
-        
+
         Arguments:
 
             api_name (str): API name to discover. *e.g.: "youtube"*
-            
+
             api_version (str): API version to discover *e.g.: "v3" not "3" and not 3*
 
             validate (bool): Set this to True to use this lib's built in parameter validation logic. Note that you shouldn't rely on this for critical user input validation.
-            
+
+            api_endpoint_url_fmt (str):
```

Comment:
Passing a string formatter seems to me like extra unnecessary work, as opposed to the user just specifying the URL version. We might as well just ask for the full URL. 

Either way, I think from an API design perspective, it would be even better if Aiogoogle automatically figures out the URL version of the passed API name and version on its own.

Either by a map of Google APIs names and versions to discovery URL versions supported OR trial and error like what the official lib does. i.e. try fetching using discovery service URL v1, if unavailable, try v2.

As a good middle ground, maybe we can just rename this parameter to just `url` and make the first parameter `api_name` optional. What do you think?

Reply:
I like your idea of allowing the user to specify [v2](https://github.com/googleapis/google-api-python-client/blob/main/googleapiclient/discovery.py#L90) (for folks who want to avoid a potentially unnecessary network call), with a fallback being to do the [trail & error method](https://github.com/googleapis/google-api-python-client/blob/1a8c294822108c82816776fe216c197754db51aa/googleapiclient/discovery.py#L283-L314) like what official lib does.

I can redo this diff to go with your suggestions 👍🏿

I am wondering though, should the constructor for the [Aiogoogle](https://github.com/omarryhan/aiogoogle/blob/ea167bac744197f8c015367a7f798365387a2773/aiogoogle/client.py#L14) client be updated to allow users to specify which discovery doc version they want to use instead of the `Aiogoogle.discover` method (defaulting to `v1` if they don't specify anything)?

I'd just add the JSON for the v2 Discovery Service doc to `aiogoogle/data.py`, and do something like this in [line 79](https://github.com/omarryhan/aiogoogle/blob/ea167bac744197f8c015367a7f798365387a2773/aiogoogle/client.py#L79) of `aiogoogle/client.py`
```
if disco_doc_ver not in {'v1, v2'}:
  raise NotImplementedError(f"Google Discovery Service version not supported: {disco_doc_ver}")
if disco_doc_ver == 'v1':
  disco_doc = DISCOVERY_SERVICE_V1_DISCOVERY_DOC
elif disco_doc_ver == 'v2':
    disco_doc = DISCOVERY_SERVICE_V2_DISCOVERY_DOC
GoogleAPI(disco_doc)
```

Reply:
Yeah that'd be more flexible *and* less intrusive of a change to the existing codebase.

I'm most likely going to explore updating the diff to add a *disco_doc_ver* param to the `discover` method, and keeping things simple by taking the [V2_DISCOVERY_URI](https://github.com/googleapis/google-api-python-client/blob/e767fa7be319213cf54381693698927df6882bbe/googleapiclient/discovery.py#L90) format string from the official Python SDK and making it a constant in Aiogoogle (and the `discover` method will fetch the API disco doc via the Google API name and version)